rct229/rules/section5/section5rule34.py

```diff
@@ -0,0 +1,107 @@
+from rct229.rule_engine.rule_base import (
+    RuleDefinitionBase,
+    RuleDefinitionListIndexedBase,
+)
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.ruleset_functions.get_building_segment_skylight_roof_areas_dict import (
+    get_building_segment_skylight_roof_areas_dict,
+)
```

Comment:
I believe that instead of checking individual segments, the RDS needs to sum up all roof areas and skylight areas returned by the function for individual building_segments to get the total skylight to roof ratio for the project (which may include multiple buildings), and then apply the RDS logic based on this total ratio.

Reply:
That seems like the right approach but the current RDS is written using by building_segment.
Do you want to bring this up to the RDS meeting? Or else we can modify the code and RDS to reflect the change.

Reply:
@weilixu I believe we discussed it at the last call and agreed that it should be aggregated to include all buildings and building segments. @supriyagoel can you please confirm?